### ISSUE TITLE
Add cross-reference for Glucose intolerance HP:0001952 to MeSH:D018149

### ISSUE THREAD

**Issue by dhimmel** [obophenotype/human-phenotype-ontology#6370]
Glucose intolerance (`HP:0001952`, https://hpo.jax.org/app/browse/term/HP:0001952) currently has a single cross-reference to `UMLS:C0235401`.

Would it be possible to also cross-reference the MeSH term `D018149` for [Glucose Intolerance](https://meshb.nlm.nih.gov/record/ui?ui=D018149)? Or are users expected to expand UMLS to obtain MeSH cross-references?

**Comment by dhimmel** [comment]
`HP:0001952` replaced `HP:0000833` which I believe did include the MeSH cross-reference (or at least does in EFO).

**Comment by dhimmel** [comment]
> dhimmel added the synonyms label

Wasn't sure which label to put this under, so used synonyms although it's not ideal.

**Comment by pnrobinson** [comment]
Sorry for the late response @dhimmel  -- we did a little revision and https://hpo.jax.org/app/browse/term/HP:0001952 does include this UMLS reference.
In 2021, we are planning on improving the way we present mappings to UMLS -- thanks to the work of the UMLS team there is a detailed mapping, but this is not straightforward to access.

**Comment by dhimmel** [comment]
Thanks @pnrobinson for the reply. My issue is about the missing cross-reference to `MESH:D018149`, not the existing cross-reference to `UMLS:C0235401`. If you could take a look at my initial comment again here that would be helpful.

**Comment by pnrobinson** [comment]
Sorry, I see what you mean. I think we are moving to the UMLS mapping, it would be a duplication of effort for us to maintain mappings ourselves to MeSH and the many other vocabularies in UMLS. This year we will be trying to offer an external file with all of these mappings that is derived from UMLS and easier to use than UMLS resources directly. Any suggestions as to what would be most useful are welcome @matentzn @drseb

**Comment by dhimmel** [comment]
> I think we are moving to the UMLS mapping, it would be a duplication of effort for us to maintain mappings ourselves to MeSH and the many other vocabularies in UMLS.

That makes sense. So that means HP will (or already has) removed all xrefs that are implied by a UMLS xref?

> This year we will be trying to offer an external file with all of these mappings that is derived from UMLS and easier to use than UMLS resources directly.

That would be helpful. In my case, I'm most interested in the cross-references for HP terms that get imported to EFO. So perhaps EFO could explode UMLS xrefs for HP terms.

I think there are two competing objectives here:

1. for ontology maintenance, it's best if a single cross-reference (like to UMLS) can be set, from which more extensive cross-references can be derived.
2. users benefit from the most extensive and complete cross-references (as long as accuracy is maintained).

So the larger question here is when should xref explosion occur, such that ontologies can maintain a single xref source but users can benefit from comprehensive xrefs?

**Comment by pnrobinson** [comment]
Reopening this for comments, @matentzn @drseb @cmungall @mellybelly  @iimpulse -- I think it would be great if we could pull in the UMLS synonyms, pass along any needed corrections to the UMLS team, and then offer a file for download as well as perhaps an API. I guess this would mean an ingest script for the UMLS releases once every six months. This would be in scope for the new HPO proposal assuming it get's funded, perhaps it is also in scope for PF

**Comment by matentzn** [comment]
Related to https://github.com/obophenotype/human-phenotype-ontology/issues/6513

I agree it would be good to solve this.

**Comment by drseb** [comment]
Agree that this is high priority. @pnrobinson do you remember our contact persons at UMLS from 1 or 2 years ago?

**Comment by pnrobinson** [comment]
I am going to move this discussion to Slack, I think this issue is closable.

**Comment by dhimmel** [comment]
> I am going to move this discussion to Slack

Did you want to drop a slack link here so people with access to that Slack can follow the conversation from here?

**Comment by matentzn** [comment]
Its a closed slack @dhimmel but we will keep you posted as we move forward!
